nautobot_golden_config/jobs.py

```diff
@@ -22,11 +22,28 @@
 name = "Golden Configuration"  # pylint: disable=invalid-name
 
 
-def get_refreshed_repos(job_obj, repo_type):
+def get_refreshed_repos(job_obj, repo_type, devices=None, gc_settings=None):
     """Small wrapper to pull latest branch, and return a GitRepo plugin specific object."""
-    repository_records = set(
-        getattr(gcs, repo_type) for gcs in GoldenConfigSetting.objects.all() if getattr(gcs, repo_type, None)
-    )
+    if isinstance(devices, Device):
+        # AllGoldenConfig only accepts a single device object, so we cast it to a list here for consistency
+        devices = [devices]
+    gc_settings_query = gc_settings if gc_settings else GoldenConfigSetting.objects.all()
+
+    if devices and all(isinstance(d, Device) for d in devices):
```

Comment:
is the `all` clause even needed?  When would an index of devices ever not be a type Device? The job input form would only allow you select device obj.

Reply:
And the `if devices` check would validate the list is at least 1 element.